src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -584,6 +590,11 @@ protected override void ComputeDependencyNodeDependencies(List<DependencyNodeCor
                             corInfoImpl.CompileMethod(methodCodeNodeNeedingCode, Logger);
                         }
                     }
+                    catch (Internal.TypeSystem.TypeSystemException.TypeLoadException)
+                    {
+                        if (Logger.IsVerbose)
+                            Logger.Writer.WriteLine($"Warning: Method `{method}` was not compiled because it participates in a generic cycle");
+                    }
```

Comment:
I would just let it fall through to the `catch (TypeSystemException ex)` below. `TypeLoadException` can signify other kind of problem. It doesn't have to be a generic cycle.

Reply:
Good suggestion, thanks, fixed in the latest commit.